### PR TITLE
Add optional limit to searchimagescmd

### LINKED ISSUE #1323: `SearchImagesCmd` should support setting a limit
See #989

### PR DESCRIPTION
**Pull request by jarebudev**
fixes #1323

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/1332)

### CODE REVIEW

File: docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java [hunk 1]
```diff
@@ -23,11 +23,22 @@ public SearchImagesCmdExec(WebTarget baseResource, DockerClientConfig dockerClie
...
+        Integer limit = command.getLimit();
+        if (limit == null) {
+            webResource = getBaseResource()
+                .path("/images/search")
+                .queryParam("term", command.getTerm());
+        } else {
+            webResource = getBaseResource()
+                .path("/images/search")
+                .queryParam("term", command.getTerm())
+                .queryParam("limit", command.getLimit());
```
Comment (KostyaSha) [inline docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java]: you can just add if check without duplication
```
        if (nonNull(limit)) {
            webResource = webResource.queryParam("limit", command.getLimit());
        }   
```

Comment (jarebudev) [reply docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java]: thanks @KostyaSha, have made change

File: docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java [hunk 2]
```diff
@@ -23,11 +23,16 @@ public SearchImagesCmdExec(WebTarget baseResource, DockerClientConfig dockerClie
 
     @Override
     protected List<SearchItem> execute(SearchImagesCmd command) {
-        WebTarget webResource = getBaseResource().path("/images/search").queryParam("term", command.getTerm());
+        WebTarget webResource = getBaseResource().path("/images/search")
+            .queryParam("term", command.getTerm());
+
+        if (command.getLimit() != null) {
+            webResource = webResource.queryParam("limit", command.getLimit());
+
```
Comment (KostyaSha) [inline docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java]: pls remove extra line

Comment (jarebudev) [reply docker-java-core/src/main/java/com/github/dockerjava/core/exec/SearchImagesCmdExec.java]: i think i can manage that :)

**Pull request merged**